src/main/java/io/vertx/cassandra/ResultSet.java

```diff
@@ -15,21 +15,68 @@
  */
 package io.vertx.cassandra;
 
+import com.datastax.driver.core.ColumnDefinitions;
 import com.datastax.driver.core.Row;
+import io.vertx.codegen.annotations.Fluent;
 import io.vertx.codegen.annotations.GenIgnore;
 import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+
+import java.util.List;
 
 /**
  * It is like {@link com.datastax.driver.core.ResultSet}, but adopted for Vert.x.
  */
 @VertxGen
-public interface ResultSet extends Iterable<Row> {
+public interface ResultSet {
+
+  /**
+   * @see com.datastax.driver.core.ResultSet#isExhausted()
+   */
+  boolean isExhausted();
+
+  /**
+   * @see com.datastax.driver.core.ResultSet#isFullyFetched()
+   */
+  boolean isFullyFetched();
+
+  /**
+   * @see com.datastax.driver.core.ResultSet#getAvailableWithoutFetching()
+   */
+  int getAvailableWithoutFetching();
 
+  /**
+   * @param handler handler called when result is fetched
+   * @see com.datastax.driver.core.ResultSet#fetchMoreResults()
+   */
+  @Fluent
+  ResultSet fetchMoreResults(Handler<AsyncResult<Void>> handler);
```

Comment:
Why making it fluent? Is there a use case for fetching without waiting for the result?

Reply:
I have marked it as fluent because it can be fluent.

Reply:
Is there any general rule which methods should not be fluent, even if can?

Reply:
Not sure about a rule but if doesn't make sense to use the API fluently then the API shouldn't be fluent. Thus my question: what's the use case for using it fluently?

Reply:
As I know, the main advantage of fluency is verbosity reduction. 

So I can imagine the use case when you need to do this:
```java
resultSet.one(...).fetchMoreResults(...)
```

This can be useful when you don't want to consume results now, and you would like `#one(...)` call to be instant later.

It is just less werbose than:
```java
resultSet.one(...)
resultSet.fetchMoreResults(...)
```

Reply:
ok